clustering/web/undertow/src/main/java/org/wildfly/clustering/web/undertow/session/DistributableSession.java

```diff
@@ -280,6 +281,9 @@ public void invalidate(HttpServerExchange exchange) {
 
     @Override
     public String changeSessionId(HttpServerExchange exchange, SessionConfig config) {
+        if (exchange.isResponseComplete()) {
+            throw UndertowMessages.MESSAGES.responseComplete();
+        }
```

Comment:
@msfm Thanks for the feedback.

> Isn't this better to use exchange.isResponseStarted() instead of exchange.isResponseComplete()?

Yes - you are absolutely right.  We want the condition to match the implementation of ServletResponse.isCommitted().  Good catch.

> Also, I do not know if UndertowMessages is good to use here

This generally fine, so long as the context of the message is exactly the same.

> Even if it's ok to use Undertow logger, UndertowMessages.MESSAGES.responseAlreadyStarted() (The response has already been startd) or UndertowServletMessages.MESSAGES.responseAlreadyCommited() (Response already commited) might be suitable log message?

I think UndertowServletMessages.MESSAGES.responseAlreadyCommited() is most appropriate here - it is the same ISE message used for a duplicate invocation of sendRedirect(...) or sendError(...).